pkg/serviceaccount/openidmetadata.go

```diff
@@ -44,26 +47,68 @@ const (
 	JWKSPath = "/openid/v1/jwks"
 )
 
-// OpenIDMetadata contains the pre-rendered responses for OIDC discovery endpoints.
-type OpenIDMetadata struct {
-	ConfigJSON       []byte
-	PublicKeysetJSON []byte
+// OpenIDMetadataProvider returns pre-rendered responses for OIDC discovery endpoints.
+type OpenIDMetadataProvider interface {
+	GetConfigJSON() (json []byte, maxAge int, err error)
+	GetKeysetJSON() (json []byte, maxAge int, err error)
```

Comment:
Should we make this not return an error since it should always be the last working config?

```suggestion
	GetConfigJSON() (json []byte, maxAge int)
	GetKeysetJSON() (json []byte, maxAge int)
```

Reply:
done